### PR TITLE
fix: panic when dumping memory profile

### LINKED ISSUE #12254: bug: panic on memory dump
### Describe the bug

```
thread 'risingwave-main' panicked at src/compute/src/memory_management/policy.rs:132:58:
called `Result::unwrap()` on an `Err` value: Attempt to read or write `void` value, or attempt to write read-only value.
stack backtrace:
   0:     0x55bb8bb06bd7 - std::backtrace_rs::backtrace::libunwind::trace::h75b30c0cd68bc7af
                               at /rustc/62ebe3a2b177d50ec664798d731b8a8d1a9120d1/library/std/src/../../backtrace/src/backtrace/libunwind.rs:93:5
   1:     0x55bb8bb06bd7 - std::backtrace_rs::backtrace::trace_unsynchronized::h87a4581e60db2a55
                               at /rustc/62ebe3a2b177d50ec664798d731b8a8d1a9120d1/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x55bb8bb06bd7 - std::sys_common::backtrace::_print_fmt::hb61bbd4cad6c8d05
                               at /rustc/62ebe3a2b177d50ec664798d731b8a8d1a9120d1/library/std/src/sys_common/backtrace.rs:67:5
   3:     0x55bb8bb06bd7 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h78c87721a2fe0851
                               at /rustc/62ebe3a2b177d50ec664798d731b8a8d1a9120d1/library/std/src/sys_common/backtrace.rs:44:22
   4:     0x55bb8569ae9c - core::fmt::rt::Argument::fmt::h3a1f78153a55cf27
                               at /rustc/62ebe3a2b177d50ec664798d731b8a8d1a9120d1/library/core/src/fmt/rt.rs:138:9
   5:     0x55bb8569ae9c - core::fmt::write::h78df6aa1c7979fcd
                               at /rustc/62ebe3a2b177d50ec664798d731b8a8d1a9120d1/library/core/src/fmt/mod.rs:1094:21
   6:     0x55bb8bb0151e - std::io::Write::write_fmt::h51abcfccd2bfcce9
                               at /rustc/62ebe3a2b177d50ec664798d731b8a8d1a9120d1/library/std/src/io/mod.rs:1714:15
   7:     0x55bb8bb069b4 - std::sys_common::backtrace::_print::h900c04ec6229a3eb
                               at /rustc/62ebe3a2b177d50ec664798d731b8a8d1a9120d1/library/std/src/sys_common/backtrace.rs:47:5
   8:     0x55bb8bb069b4 - std::sys_common::backtrace::print::heb918b083479ba12
                               at /rustc/62ebe3a2b177d50ec664798d731b8a8d1a9120d1/library/std/src/sys_common/backtrace.rs:34:9
   9:     0x55bb8bb0847b - std::panicking::panic_hook_with_disk_dump::{{closure}}::h308105f13195c33c
                               at /rustc/62ebe3a2b177d50ec664798d731b8a8d1a9120d1/library/std/src/panicking.rs:280:22
  10:     0x55bb8bb08136 - std::panicking::panic_hook_with_disk_dump::h2f5bad6440ca252c
                               at /rustc/62ebe3a2b177d50ec664798d731b8a8d1a9120d1/library/std/src/panicking.rs:314:9
```

### Error message/log

_No response_

### To Reproduce

_No response_

### Expected behavior

_No response_

### How did you deploy RisingWave?

_No response_

### The version of RisingWave

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by fuyufjh**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Resolves #12254

See also https://github.com/risingwavelabs/jemallocator/commit/c72ce6a9a6ca070f65644d3f3d77f4633c2fde47

The bug is `dump.prefix` is [write-only](https://github.com/jemalloc/jemalloc/blob/54eaed1d8b56b1aa528be3bdd1877e59c56fa90c/src/ctl.c#L3310). Now use `opt.dump_prefix` instead.

Also resolves https://github.com/risingwavelabs/risingwave/pull/12186#discussion_r1320966334

## Checklist

- [ ] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Review by yuhao-su — APPROVED** [review]

File: src/compute/Cargo.toml [hunk 1]
```diff
@@ -55,7 +55,7 @@ tower = { version = "0.4", features = ["util", "load-shed"] }
 tracing = "0.1"
 
 [target.'cfg(target_os = "linux")'.dependencies]
-tikv-jemalloc-ctl = { git = "https://github.com/risingwavelabs/jemallocator.git", rev = "b7f9f3" }
+tikv-jemalloc-ctl = { git = "https://github.com/risingwavelabs/jemallocator.git", rev = "64a2d9" }
```
Comment (BugenZhao) [inline src/compute/Cargo.toml:58]: Also use the workspace dependency? 👀

**Review by BugenZhao — APPROVED** [review]

File: src/compute/Cargo.toml (hunk 1 above)
Comment (fuyufjh) [reply src/compute/Cargo.toml:58]: `tikv-jemalloc-ctl` is different from `tikv-jemallocator`, and it's only used in `compute` for memory controller policy. Are you sure to move it to workspace dependency as well?

Comment (yuhao-su) [reply src/compute/Cargo.toml:58]: There can never exist two different version of jemalloc ctl
/allocator. So use worksapce makes sense to me.

Comment (yuhao-su) [reply src/compute/Cargo.toml:58]: also it can be used in all nodes for profiling

Comment (BugenZhao) [reply src/compute/Cargo.toml:58]: > `tikv-jemalloc-ctl` is different from `tikv-jemallocator`, and it's only used in `compute` for memory controller policy.

Oh, I didn't realize this.

**Review by BugenZhao — APPROVED** [review]

File: src/compute/Cargo.toml (hunk 1 above)
Comment (fuyufjh) [reply src/compute/Cargo.toml:58]: > also it can be used in all nodes for profiling

Agree. Let's do it upon that time!

**Comment by fuyufjh** [comment]
Encountered a weird problem: simulation test stack overflowed. @wangrunji0408 is helping with it.

[gdb_bt.txt](https://github.com/risingwavelabs/risingwave/files/12604732/gdb_bt.txt)

**Comment by wangrunji0408** [comment]
> Encountered a weird problem: simulation test stack overflowed. @wangrunji0408 is helping with it.
> 
> [gdb_bt.txt](https://github.com/risingwavelabs/risingwave/files/12604732/gdb_bt.txt)

Found the problem from stack trace:
jemalloc init -> get system time -> intercepted by madsim -> get thread local variable -> malloc -> jemalloc init (infinite loop 💥)

I can't reproduce it on my Mac. Looks like the problem is only on Linux.

**Comment by fuyufjh** [comment]
It might be related to `unprefixed_malloc_on_supported_platforms`. Since I moved the dependency to workspace level, this is enabled globally. Let me try to disable it for simulation test. 


> Looks like the problem is only on Linux.

Yes, because unprefixed jemalloc is not available in MacOS.

**Comment by wangrunji0408** [comment]
> It might be related to `unprefixed_malloc_on_supported_platforms`.

Yes. I can reproduce it on Linux with this feature enabled. And the bug disappeared when I disabled this feature.

**Comment by BugenZhao** [comment]
> Since I moved the dependency to workspace level, this is enabled globally.

Oops. 🫨 I'm not sure about this. Is that true?

**Comment by fuyufjh** [comment]
> > It might be related to `unprefixed_malloc_on_supported_platforms`.
> 
> Yes. I can reproduce it on Linux with this feature enabled. And the bug disappeared when I disabled this feature.
> 
> I'm curious about what specific things this feature does. 🤯 https://github.com/tikv/jemallocator/blob/0f8983f71813c6e052c6ab445e965c4b0a7e251e/jemalloc-sys/README.md?plain=1#L72

It's to solve https://github.com/risingwavelabs/risingwave/issues/9669

**Pull request merged**